corehq/motech/tests/test_value_source.py

```diff
@@ -114,3 +116,33 @@ def test_null_case_property(self):
         case_property = CaseProperty.wrap({"case_property": None})
         with self.assertRaisesRegexp(BadValueError, "Property case_property is required."):
             case_property.validate()
+
+
+class ConstantStringTests(SimpleTestCase):
+
+    def test_get_value(self):
+        constant = ConstantString.wrap({"value": "foo"})
+        info = CaseTriggerInfo("test-domain", None)
+        value = constant.get_value(info)
+        self.assertEqual(value, "foo")
+
+    def test_default_get_value(self):
+        constant = ConstantString.wrap({})
+        info = CaseTriggerInfo("test-domain", None)
+        value = constant.get_value(info)
+        self.assertIsNone(value)
+
+    def test_casting_value(self):
+        constant = ConstantString.wrap({
+            "value": "1",
+            "external_data_type": COMMCARE_DATA_TYPE_DECIMAL,
+        })
+        info = CaseTriggerInfo("test-domain", None)
+        value = constant.get_value(info)
+        self.assertEqual(value, 1.0)
```

Comment:
Couldn't help but notice the apparent inconsistency between the type of this value (`float`) and the name `COMMCARE_DATA_TYPE_DECIMAL`. In Python `Decimal != float`. I realize that distinction was not created in this PR, but just mentioning in case it's close enough to the birth of that constant name to rename it to something like `COMMCARE_DATA_TYPE_FLOAT`. Or maybe I'm missing something?

Reply:
The constants were named after CommCare question types, not Python data types, because CommCare users would know the question types, and its value, "cc_decimal" would probably make more sense to them than "cc_float" to describe a value that came from a decimal form question. 

But I appreciate that there is more than one aspect here that is confusing: You mentioned Python Decimal vs float. Also, they are called "data types" not "question types" because their values don't always come from form questions -- they can also come from case properties, location metadata, and constants.

So I'm open to suggestions, but we should be careful not to replace the name with one that is confusing in a different way.